modules/xfeatures2d/src/freak.cpp

```diff
@@ -333,8 +333,11 @@ void FREAK_Impl::compute( InputArray _image, std::vector<KeyPoint>& keypoints, O
     ((FREAK_Impl*)this)->buildPattern();
 
     // Convert to gray if not already
-    Mat grayImage = image;
-    CV_Assert(grayImage.channels() == 1);
+    Mat grayImage;
+    if( image.channels() == 3 || image.channels() == 4 )
+        cvtColor(image, grayImage, COLOR_BGR2GRAY);
+    else
+        grayImage = image;
```

Comment:
There also could be image.channels() == 2 case that now will fall throw here.

Reply:
`CV_Assert()` should be restored (on line 337) to check supported number of channels (updated variant).